advisor/src/test/assets/ort-analyzer-result.yml

```diff
@@ -571,4 +571,8 @@ analyzer:
 scanner: null
 advisor: null
 evaluator: null
-resolved_configuration: {}
+resolved_configuration:
+  package_curations:
+  - provider:
+      id: "DefaultDir"
+    curations: []
```

Comment:
This is something that I believe we forgot to discuss in our meeting, but IMO providers that don't have any curations should not be listed here. This is about *resolved* (i.e. relevant) configuration after all, and configured providers that don't contribute anything are not relevant.

Reply:
On the other hand this empty entry documents that the provider was queried and did not provide any applicable curations.

Reply:
I consciously decided for listing all enabled providers, to have the priorities of the providers in the ort result.
This enables the use case for re-resolving curations for any particular provider ID, no matter if the initial resolution lead to an empty result. Therefore I have a rather strong preference to keep all enabled providers in the list.

ok with that @mnonnenmacher  @sschuberth ?

Reply:
> On the other hand this empty entry documents that the provider was queried and did not provide any applicable curations.

But what does that tell you? For example, it doesn't tell you that you could disable the provider, as it might have applicable curations for a slightly different set of packages.

Reply:
> > On the other hand this empty entry documents that the provider was queried and did not provide any applicable curations.
> 
> But what does that tell you? For example, it doesn't tell you that you could disable the provider, as it might have applicable curations for a slightly different set of packages.

It tells you that at the time the ORT result was created the provider had no curations for the packages contained in the result.

Reply:
> It tells you that at the time the ORT result was created the provider had no curations for the packages contained in the result.

Obviously, yes. But I have doubts that this is valuable information. Anyway, what convinced me more is the re-resolving use-case and to maintain the order of providers there.